data/avro-schema/src/main/resources/avro/net/corda/data/flow/output/FlowStatus.avsc

```diff
@@ -49,6 +50,17 @@
       "type": ["null", "net.corda.data.ExceptionEnvelope"],
       "doc": "Optional error message, this will be set if the flow status is 'FAILED'"
     },
+    {
+      "name": "processingTerminationDetails",
+      "type": [
+        "null",
+        {
+            "type": "map",
```

Comment:
I'm pretty sure we cannot use Avro map types because the order in which it sends the entries back is non deterministic which means we cannot sign the messages to validate authenticity (i.e. we don't know the exact content in advance at the receiving end).

This has been solved already however, there's a reusable Corda type called `KeyValuePairList` and there are some helper methods in the codebase to help you interact with it in map like ways too. E.g.
https://github.com/corda/corda-api/blob/dc4a95b7728aa9b4ac84b135395459def407d068/data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/external/ExternalEventContext.avsc#L18

Reply:
im wondering should I raise a ticket for FindWithNamedQuery

Reply:
Thanks Simon I wasn't aware of this

Reply:
I assume it applies to everything. @mnesbit or @JamesHR3 would have more information.

Reply:
I'm actually getting rid of the map, it's not necessary, a simple string is suffice